### ISSUE TITLE
VAE loss for non-binary images

### ISSUE THREAD

**Issue by malik-ali** [PyTorchLightning/pytorch-lightning-bolts#177]
## 🐛 Bug

I think the VAE ELBO loss when training on imagenet or cifar10 should use `F.mse_loss` instead of `F.binary_cross_entropy`. The tensors for the target `x` right now contain values outside of `[0, 1]` which might not behave sensibly with BCE. For nonbinary images, the Gaussian posterior would imply MSE loss.

### To Reproduce
See here:
https://github.com/PyTorchLightning/pytorch-lightning-bolts/blob/cf3830787e1352ce71d673347fb5648c90497b48/pl_bolts/models/autoencoders/basic_vae/basic_vae_module.py#L160

**Comment by akihironitta** [comment]
@malik-ali Each pixel in an image is usually in the range between 0 and 1, and I think it seems reasonable to use `F.binary_cross_entropy` as implemented here. However, sometimes we use the range -1 and 1 (or possibly other ranges) which can help learn faster and better, and I believe we cannot use `F.binary_cross_entropy` in this case because it can take logarithms of negative values.

In order to solve this, you can just define a new class by extending the original `VAE` class and override the ELBO using mse or whatever loss you want to use.

**Comment by qbeer** [comment]
@akihironitta @malik-ali He could extend the class but indeed he is right. The `binary_cross_entropy` should only be used with binarized data, it is not right for CIFAR-10 or ImageNet data. It might explain why the examples look off in color in my opinion.

**Comment by ananyahjha93** [comment]
@malik-ali @qbeer this has been fixed.
